test_conformance/api/negative_device.cpp

```diff
@@ -0,0 +1,536 @@
+            "\"the platform associated with device does not support device and "
+            "host timer synchronization\"",
+            TEST_FAIL);
+    }
+
+    return TEST_PASS;
+}
+
+/* Negative Tests for clGetHostTimer */
+int test_negative_get_host_timer(cl_device_id deviceID, cl_context context,
+                                 cl_command_queue queue, int num_elements)
+{
+    cl_ulong *host_timestamp = nullptr;
+    cl_int err = clGetHostTimer(nullptr, host_timestamp);
+    test_failure_error_ret(
+        err, CL_INVALID_DEVICE,
+        "clGetHostTimer should return CL_INVALID_DEVICE when: \"device is not "
+        "a valid device\" using a nullptr",
+        TEST_FAIL);
+
+    err =
+        clGetHostTimer(reinterpret_cast<cl_device_id>(context), host_timestamp);
+    test_failure_error_ret(
+        err, CL_INVALID_DEVICE,
+        "clGetHostTimer should return CL_INVALID_DEVICE when: \"device is not "
+        "a valid device\" using a valid object which is NOT a device",
+        TEST_FAIL);
+
+    cl_platform_id platform = getPlatformFromDevice(deviceID);
+    // Initialise timer_resolution to a Non-0 value as CL2.1/2 devices must
+    // support timer synchronisation
+    cl_ulong timer_resolution = 1;
+    if (get_device_cl_version(deviceID) >= Version(3, 0))
+    {
+        err = clGetPlatformInfo(platform, CL_PLATFORM_HOST_TIMER_RESOLUTION,
+                                sizeof(timer_resolution), &timer_resolution,
+                                nullptr);
+        test_error(err, "clGetPlatformInfo");
+    }
+
+    if (timer_resolution != 0)
+    {
+        log_info("Platform Supports Timers\n");
+        log_info("Skipping CL_INVALID_OPERATION tests\n");
+
+        err = clGetHostTimer(deviceID, nullptr);
+        test_failure_error_ret(err, CL_INVALID_VALUE,
+                               "clGetHostTimer should return CL_INVALID_VALUE "
+                               "when: \"host_timestamp is NULL\"",
+                               TEST_FAIL);
+    }
+    else
+    {
+        log_info("Platform does not Support Timers\n");
+        log_info("Skipping CL_INVALID_VALUE tests\n");
+
+        err = clGetHostTimer(deviceID, host_timestamp);
+        test_failure_error_ret(
+            err, CL_INVALID_OPERATION,
+            "clGetHostTimer should return CL_INVALID_OPERATION when: \"the "
+            "platform associated with device does not support device and host "
+            "timer synchronization\"",
+            TEST_FAIL);
+    }
+
+    return TEST_PASS;
+}
+
+/* Negative Tests for clCreateSubDevices */
+enum SupportedPartitionSchemes
+{
+    None = 0,
+    Equally = 1 << 0,
+    Counts = 1 << 1,
+    Affinity = 1 << 2,
+    All_Schemes = Affinity | Counts | Equally,
+};
+
+static int get_supported_properties(cl_device_id deviceID)
+{
+    size_t number_of_properties = 0;
+    int err = clGetDeviceInfo(deviceID, CL_DEVICE_PARTITION_PROPERTIES, 0,
+                              nullptr, &number_of_properties);
```

Comment:
It's the total size of the data that's returned, you need to divide by `sizeof(cl_device_partition_property)` to get the number of properties.